File: 07_app_development/MyProject/libtsparser/src/main/java/com/tosmart/xiexuming/libtsparser/table/ProgramAssociationTable.java
package com.tosmart.xiexuming.libtsparser.table;

import androidx.annotation.NonNull;

import com.tosmart.xiexuming.libtsparser.program.Program;

import java.util.ArrayList;

/**
 * @author xiexuming
 */
public class ProgramAssociationTable {
    private ArrayList<Program> programs;
    private int version;

    @NonNull
    @Override
    public String toString() {
        return "ProgramAssociationTable{" +
                " version=" + version +
                ", programs=" + programs +
                '}';
    }

    public ArrayList<Program> getPrograms() {
        return programs;
    }

    public void setPrograms(ArrayList<Program> programs) {
        this.programs = programs;
    }

    public int getVersion() {
        return version;
    }

    public void setVersion(int version) {
        this.version = version;
    }
}

File: newpaserapp/libtosmartparse/src/main/java/com/example/libtosmartparse/table/ProgramAssociationTable.kt
package com.example.libtosmartparse.table

import Program

/**
 * @author xxm
 * 2022/7/17
 **/
class ProgramAssociationTable {
    private var programs: ArrayList<Program>? = null
    private var version = 0
    override fun toString(): String {
        return "ProgramAssociationTable{" +
                " version=" + version +
                ", programs=" + programs +
                '}'
    }

    fun getPrograms(): ArrayList<Program>? {
        return programs
    }

    fun setPrograms(programs: ArrayList<Program>?) {
        this.programs = programs
    }

    fun getVersion(): Int {
        return version
    }

    fun setVersion(version: Int) {
        this.version = version
    }
}
